portfolio/src/main/java/com/google/sps/servlets/VKDataServlet.java

```diff
@@ -0,0 +1,162 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.servlets;
+
+import com.google.gson.Gson;
+import java.io.IOException;
+import java.util.List;
+import java.util.ArrayList;
+import java.util.Map;
+import java.util.HashMap;
+import java.util.LinkedHashMap;
+import java.util.Comparator;
+import java.util.Scanner;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@WebServlet("/vk-data")
+public class VKDataServlet extends HttpServlet {
+
+  private Map<String, Integer> countries_sorted = new LinkedHashMap<>();
+  private Map<String, Integer> cities_sorted = new LinkedHashMap<>();
+  private Map<String, Integer> genders = new HashMap<>();
+  private Map<String, Map<String, Integer>> views = new LinkedHashMap<>();
+
+  @Override
+  public void init() {
+    Map<String, Integer> countries = new HashMap<>();
+    Map<String, Integer> cities = new HashMap<>();
+
+    Scanner scanner = new Scanner(getServletContext().getResourceAsStream(
+        "/WEB-INF/vkontakte_stat.csv"));
+    while (scanner.hasNextLine()) {
+      String line = scanner.nextLine();
+      String[] cells = line.split(",(?=([^\"]*\"[^\"]*\")*[^\"]*$)");
+      String year = cells[0].replace("\"", "").split("\\.")[2];
+      String type = cells[1].replace("\"", "");
+      switch (type) {
+        case "countries":
+          String country = cells[2].replace("\"", "");
+          try {
+            Integer value = Integer.valueOf(cells[4]);
+            Integer current_value = countries.get(country);
+            if (!country.equals("Russia") && !country.equals("")) {
+              countries.put(country, current_value == null ? value : current_value + value);
+            }
+          } catch (NumberFormatException e) {
+            System.out.println(e);
+          }
+          break;
```

Comment:
You can simplify this switch/case statement:
Do the key/value extraction before:
```
String key = cells[2];
if(key.isEmpty()) {continue;}
Integer value = 0;
try {
value = Integer.valueOf(cells[4]);
} catch() {
...
continue;
}
if(value == 0) { continue; }
```

Then a simple case statement:
```
switch() {
case "countries":
   countries.put(key, countries.getOrDefault(key, 0) + value);
}
```